test/index.js

```diff
@@ -269,15 +269,56 @@ test.serial('function() verifies cache with shouldRevalidate callback', async t
 	t.is(spy.callCount, 1);
 });
 
-test.serial('function() avoids duplicate nearby function calls', async t => {
+test.serial('function() avoids concurrent function calls', async t => {
 	const spy = sinon.spy(getUsernameDemo);
 	const call = cache.function(spy);
 
 	t.is(spy.callCount, 0);
-	const cacheMePlease = function () {};
-	t.is(call('@anne', cacheMePlease), call('@anne', cacheMePlease));
-	await call('@anne', cacheMePlease);
+	t.is(call('@anne'), call('@anne'));
+	await call('@anne');
 	t.is(spy.callCount, 1);
+
+	t.not(call('@new'), call('@other'));
+	await call('@idk');
+	t.is(spy.callCount, 4);
+});
+
+test.serial.failing('function() avoids concurrent function calls with complex arguments', async t => {
+	const spy = sinon.spy(async (transform, user) => transform(user.name));
+	const call = cache.function(spy);
```

Comment:
This is actually something that probably should have never worked without `cacheKey` and should probably be updated instead of left as `failing`

```suggestion
test.serial.failing('function() avoids concurrent function calls with complex arguments', async t => {
	const spy = sinon.spy(async (transform, user) => transform(user.name));
	const call = cache.function(spy, {cacheKey: ([fn, user]) => JSON.stringify([fn.name, user])});
```